src/app/asset-page/artstor-viewer/artstor-viewer.component.ts

```diff
@@ -406,6 +406,29 @@ export class ArtstorViewerComponent implements OnInit, OnDestroy {
             this.osdViewer.destroy();
         });
 
+        this.osdViewer.addOnceHandler("tile-drawn", () => {
+            // Get all canvas elements, first one is the main canvas, second one is the navigator
```

Comment:
I think we omit the navigator when in full screen mode. As a result, our `canvasElements` might not always match the ordering we are expecting them to be.

Overall I am a little concerned with how fragile this tab logic can be. I don't have any great ideas at the moment, but I would love to brainstorm some with you!

Reply:
Oh yes we omit the navigator in full screen mode. But this event is fired only once when the tiles are drawn, which is when we first land on this asset page. So by the time we open fullscreen, the event listeners have already been attached to those elements.

Reply:
Just off the top of my head, is it possible to give the elements that need the listeners a different class so you can look for that and don't have to rely on `openseadragon-canvas`?

Reply:
Hmm wonder if we will encounter any issues if the user opens up the viewer directly from an image group

Reply:
@eslawski I just had another look, so in fullscreen mode, although we don't show the navigator, it still exists in the dom, so it won't mess up the order. I would be up for a better solution though.

Reply:
@gbassham The problem is (I think), those elements are generated by openseadragon, I don't have control over them.

Reply:
Yeah, I was afraid you were going to say that.

Reply:
I was playing around with this a bit and I think I might have come up with something that prevents us from having to listen to the "tab" key. Instead I just leverage the `tabindex` so all thumbnails are tabbable to. I still needed to listen for the `tile-drawn` event, not sure there is a way around that. Let me know if this makes sense.

```        this.osdViewer.addOnceHandler("tile-drawn", () => {
            const referenceStripElements = document.querySelectorAll(".referencestrip .openseadragon-canvas");
            referenceStripElements.forEach((thumbnail, index) => {
                thumbnail.setAttribute("tabindex", "0");
                thumbnail.addEventListener("keydown", (event: KeyboardEvent) => {
                    if (event.keyCode === 13) {
                        this.osdViewer.goToPage(index)
                    }
                })
            })
        })```

Reply:
One reason that I'm listening to the tab key is that we have to jump between "artstor-viewer" component and "asset-page" component. Normally (if I don't set up the listeners), we will be tabbing through all the interactive elements in one component before we can ever reach anything in the second.

Reply:
Oh my previous comment is under the assumption that we have all interactive elements with `tabIndex=0`, but if we set them to be `tabIndex=1`, `tabIndex=2`, etc, we might be able to accomplish some particular ordering. Worth trying.